File: src/modelos/Carta.ts
export class Carta {
  cor: string;
  valor: string;
  imagem: string;
  acao?: () => void;

  constructor(cor: string, valor: string, acao?: () => void) {
    this.cor = cor;
    this.valor = valor;
    this.imagem = this.cardImage(cor, valor);
    this.acao = acao;
  }

  private cardImage = (cor: string, valor: string) => `src/assets/${cor}/${valor}.png`;
}

File: uno-server/src/main/java/com/usjt/a3/unogame/modelo/Carta.java
package com.usjt.a3.unogame.modelo;

import java.util.Random;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class Carta {
    private String id;
    private String cor;
    private String valor;
    private String imagem;
    private Runnable acaoCarta;

    public Carta(String cor, String valor) {
        Random random = new Random();

        this.id = String.format("%s-%s-%d", cor, valor, random.nextInt());
        this.cor = cor;
        this.valor = valor;
        this.imagem = String.format("src/assets/%s/%s.png", cor, valor);
    }

    public Carta(String cor, String valor, Runnable acaoCarta) {
        this(cor, valor);
        this.acaoCarta = acaoCarta;
    }
}